src/structstore_pybind.cpp

```diff
@@ -257,6 +295,17 @@ void register_structstore_pybind(py::module_& m) {
         auto lock = list.read_lock();
         return to_object<false>(list[index].get_field());
     });
+    list.def("__delitem__", [](List& list, size_t index) {
+        auto lock = list.write_lock();
+        list.erase(index);
+    });
+    list.def("__iter__", [](List& list) {
+        auto lock = list.read_lock();
+        return pybind11::make_iterator(list.begin(), list.end());
```

Comment:
Do we need some more locking here in the future? This might horribly fail in case another process manipulates the list while this iterator is still used

Reply:
Yes this will fail horribly and more locking is required for sure. Maybe we can tie the lock lifetime to the iterator lifetime?

Reply:
Refer to issue #10 for further development.